### ISSUE TITLE
ENH/FIX: croping will leave out samples if .times vector includes some rounding errors, etc.

### ISSUE THREAD

**Issue by dengemann** [mne-tools/mne-python#1811]
imagine you have 1 time sample -0.004000000000000000000036 that you compare against tmin -0.004

the croping code will use naive comparison against tmin and tmax:

instead of

```Python
tmask = (self.times >= tmin) & (self.times <= tmax)```
```
use 

```Python
tmask = (((self.times > tmin) | np.isclose(self.times, tmin)) &
         ((self.times < tmax) | np.isclose(self.times, tmax)))
```

We should probably have utils function that generalizes and unifies time sample selection.

we need to do an exclusive less / greater in conjunction with np.isclose to avoid leaving out samples 

FYI this has been an issue when comparing results between pipelines.

I'm sure several parts of our code are affected by this issue.

cc @Eric89GXL @agramfort @fraimondo

**Comment by agramfort** [comment]
sounds legit to me.

we should list the places that need this change

**Comment by Eric89GXL** [comment]
+1 for doing this as a fix for 0.9. I might have time this week to tackle this if it seems worthwhile.

**Comment by agramfort** [comment]
+1
